agent/config/parse_windows.go

```diff
@@ -114,3 +141,42 @@ var IsWindows2016 = func() (bool, error) {
 
 	return isWS2016, nil
 }
+
+var queryDomainlessGmsaPluginSubKeys = func() ([]string, error) {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SYSTEM\CurrentControlSet\Control\CCG\COMClasses\`, registry.READ)
+	if err != nil {
+		seelog.Errorf("Failed to open registry key SYSTEM\\CurrentControlSet\\Control\\CCG\\COMClasses with error: %v", err)
+		return nil, err
+	}
+	defer k.Close()
+	stat, err := k.Stat()
+	if err != nil {
+		seelog.Errorf("Failed to stat registry key SYSTEM\\CurrentControlSet\\Control\\CCG\\COMClasses with error: %v", err)
+		return nil, err
+	}
+	subKeys, err := k.ReadSubKeyNames(int(stat.SubKeyCount))
+	if err != nil {
+		seelog.Errorf("Failed to read subkeys of SYSTEM\\CurrentControlSet\\Control\\CCG\\COMClasses with error: %v", err)
+		return nil, err
+	}
+
+	seelog.Debugf("gMSA Subkeys are %+v", subKeys)
+	return subKeys, nil
+}
+
+// This function queries all gmsa plugin subkeys to check whether the Amazon ECS Plugin GUID is present.
+func isDomainlessGmsaPluginInstalled() (bool, error) {
+	subKeys, err := fnQueryDomainlessGmsaPluginSubKeys()
```

Comment:
Should we also check for the presence of GMSA binary?

Reply:
I had asked Justin about this, and he had only wanted to check for the presence of registry key (which would imply plugin is installed)